static/app/components/events/interfaces/spans/anchorLinkManager.tsx

```diff
@@ -19,14 +19,23 @@ export class Provider extends Component<Props> {
     this.scrollToHash(location.hash);
   }
 
-  scrollFns: Map<string, () => void> = new Map();
+  scrollFns: Map<string, {fn: () => void; isSpanInGroup: boolean}> = new Map();
 
   scrollToHash = (hash: string) => {
-    this.scrollFns.get(hash)?.();
+    if (this.scrollFns.has(hash)) {
+      const {fn, isSpanInGroup} = this.scrollFns.get(hash)!;
+      fn();
+
+      // If the anchored span is part of a group, need to call scrollToHash again, since the initial fn() call will only expand the group.
+      // The function gets registered again after the group is expanded, which will allow the page to scroll to the span
+      if (isSpanInGroup) {
+        setTimeout(() => this.scrollFns.get(hash)?.fn());
```

Comment:
I know this solution seems quite hacky, but I could not find any other way to get this to work properly 😥

I'm using `setTimeout` to ensure that the next function is added to the job queue, which means that it will be called after the call stack is empty. This ensures that `scrollFns.get(hash)?.fn())` is called after the span group is expanded and the changes are rendered (handled by the previous call to `fn()`).

In `spanBar.tsx`, after the component is rendered, it re-registers the function with it's span ID hash. This is the function that is called by the `setTimeout` callback here.